test/transactions/compound-v3-leverage.test.ts

```diff
@@ -0,0 +1,157 @@
+      '0x0d500B1d8E8eF31E21C99d1Db9A6444d3ADf1270': '0xe7804c37c13166ff0b37f5ae0bb07a3aebb6e245', // WMATIC
+    },
+  },
+};
+
+async function claimToken(
+  chainId: number,
+  recepient: string,
+  tokenOrAddress: common.TokenOrAddress,
+  amount: string,
+  provider: providers.Provider
+) {
+  const hre = await import('hardhat');
+
+  const web3Toolkit = new common.Web3Toolkit(chainId, /*hre.ethers.provider*/ provider);
+  const token = await web3Toolkit.getToken(tokenOrAddress);
+  const tokenAmount = new common.TokenAmount(token, amount);
+
+  let faucet: string;
+  if (token.isNative || token.isWrapped) {
+    const signers = await hre.ethers.getSigners();
+    faucet = signers[signers.length - 1].address;
+  } else {
+    faucet = faucetMap[chainId]?.specified?.[token.address] ?? faucetMap[chainId].default;
+    await helpers.impersonateAccount(faucet);
+  }
+
+  const signer = await hre.ethers.provider.getSigner(faucet);
+  if (token.isNative) {
+    await signer.sendTransaction({ to: recepient, value: tokenAmount.amountWei });
+  } else {
+    if (token.isWrapped) {
+      if (chainId == 1) {
+        const weth = common.WETH__factory.connect(token.address, signer);
+        await (await weth.deposit({ value: tokenAmount.amountWei })).wait();
+      } else if (chainId == 137) {
+        const wmatic = common.WETH__factory.connect(token.address, signer);
+        await (await wmatic.deposit({ value: tokenAmount.amountWei })).wait();
+      }
+    }
+    const erc20 = common.ERC20__factory.connect(token.address, signer);
+    await (await erc20.transfer(recepient, tokenAmount.amountWei)).wait();
+  }
+}
+
+describe('Transaction: CompoundV3', function () {
+  let chainId: number;
+  let network: common.Network;
+  let provider: providers.Provider;
+
+  let user1: SignerWithAddress;
+  let leverageToken: common.Token;
+  let leverageTokenFullBalance: string;
+  let borrowToken: common.Token;
+  let borrowTokenFullBalance: string;
+  let marketIdUSDC: string;
+
+  before(async function () {
+    chainId = common.ChainId.polygon;
+    network = common.getNetwork(chainId);
+    provider = hre.ethers.provider;
+    [, user1] = await hre.ethers.getSigners();
+    leverageToken = logics.compoundv3.polygonTokens.WMATIC;
+    leverageTokenFullBalance = '500';
+    borrowToken = logics.compoundv3.polygonTokens.USDC;
+    borrowTokenFullBalance = '500';
+    marketIdUSDC = logics.compoundv3.MarketId.USDC;
+    await claimToken(chainId, user1.address, leverageToken, leverageTokenFullBalance, provider);
+    await claimToken(chainId, user1.address, borrowToken, borrowTokenFullBalance, provider);
+  });
+
+  snapshotAndRevertEach();
+
+  it('user leverage Wrapped Native Token in USDC market', async function () {
+    // 1. Get logics
+    // 1-0. supply leverage token to USDC market
+    let initSupplyAmount = leverageTokenFullBalance;
+    let leverageAmount = '200';
```

Comment:
The current leverage amount does not take the token price into consideration. Do we need to do that to prevent the test from failing?

Reply:
After discussing with @chouandy, we would verify the user collateral amount.